guvnor-ala/guvnor-ala-ui/guvnor-ala-ui-backend/src/main/java/org/guvnor/ala/ui/backend/service/RuntimeListItemBuilder.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.ui.backend.service;
+
+import org.guvnor.ala.services.api.RuntimeQueryResultItem;
+import org.guvnor.ala.ui.model.Pipeline;
+import org.guvnor.ala.ui.model.PipelineExecutionTrace;
+import org.guvnor.ala.ui.model.PipelineExecutionTraceKey;
+import org.guvnor.ala.ui.model.PipelineKey;
+import org.guvnor.ala.ui.model.PipelineStatus;
+import org.guvnor.ala.ui.model.ProviderKey;
+import org.guvnor.ala.ui.model.ProviderTypeKey;
+import org.guvnor.ala.ui.model.Runtime;
+import org.guvnor.ala.ui.model.RuntimeKey;
+import org.guvnor.ala.ui.model.RuntimeListItem;
+import org.guvnor.ala.ui.model.RuntimeStatus;
+import org.guvnor.ala.ui.model.Stage;
+
+/**
+ * Helper class for building RuntimeListItems.
+ */
+public class RuntimeListItemBuilder {
+
+    private static final String SCHEDULED = "SCHEDULED";
+
+    private static final String RUNNING = "RUNNING";
+
+    private static final String FINISHED = "FINISHED";
+
+    private static final String ERROR = "ERROR";
+
+    private static final String STOPPED = "STOPPED";
```

Comment:
same as above comments, basically to avoid sending enums along the wire. The runtime state in the original ALA api is also a string. But we can review this when we add the runtime operations in the UI in next Sprint, or really current sprint. But I have a chain PRs, changes, pending, reviews, a new PR wating for this PR, etc, that it's a nightmare to touch this right now. But can be reviewed.